### PR TITLE
Iclr19/fix reply email

### PR DESCRIPTION
**Pull request by mohituniyal**
1. commentProcess.js : Added reply_mail which is sent when someone replies to a non-anonymous comment or a review. If the comment is anonymous then there is no way to find the id of the tauthor.
2. Added some corrections to previous code.

### CODE REVIEW

**Comment by mohituniyal** [comment]
Thanks @zbialecki . These changes have been made.

File: venues/ICLR.cc/2019/Conference/process/commentProcess.js [hunk 1]
```diff
@@ -70,6 +61,15 @@ function(){
         promises.push(or3client.or3request(or3client.mailUrl, ac_mail, 'POST', token));
       }
 
+      if(note.replyto != note.forum && replytoNoteSignatures != '(anonymous)'){
```
Comment (melisabok) [inline venues/ICLR.cc/2019/Conference/process/commentProcess.js]: I would suggest to send the emails with the following rules:


```
if note.forum == note.replyto

      if note.signatures[0].endsWith('/Authors')
            mailto: Authors: "An author posted a comment to a paper your are a co-author. "
            mailto: Reviewers: "An author posted a comment to a paper your are reviewing. "
            mailto: AreaChairs: "An author posted a comment to a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "An author posted a comment to a paper. "

      if note.signatures[0].endsWith('/AnonReviewer.*')
            mailto: Authors: "A reviewer posted a comment to a paper your are a co-author. "
            mailto: Reviewers: "A reviewer posted a comment to a paper your are reviewing. "
            mailto: AreaChairs: "A reviewer posted a comment to a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "A reviewer posted a comment to a paper. "

      if note.signatures[0].endsWith('/Area_Chairs')
            mailto: Authors: "An area chair posted a comment to a paper your are a co-author. "
            mailto: Reviewers: "An area chair posted a comment to a paper your are reviewing. "
            mailto: AreaChairs: "An area chair posted a comment to a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "An area chair posted a comment to a paper. "

      if note.signatures[0].endsWith('/Program_Chairs')
            mailto: Authors: "A program chair posted a comment to a paper your are a co-author. "
            mailto: Reviewers: "A program chair posted a comment to a paper your are reviewing. "
            mailto: AreaChairs: "A program chair posted a comment to a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "A program chair posted a comment to a paper. "

      else 
            mailto: Authors: "A comment was posted to a paper your are a co-author. "
            mailto: Reviewers: "A comment was posted to a paper your are reviewing. "
            mailto: AreaChairs: "A comment was posted to a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "A comment was posted to a comment to a paper. "

if note.forum != note.replyto

      if note.signatures[0].endsWith('/Authors')
            mailto: Authors: "An author replied a comment from a paper your are a co-author. "
            mailto: Reviewers: "An author replied a comment from a paper your are reviewing. "
            mailto: AreaChairs: "An author replied a comment from a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "An author replied a comment from a paper. "

      if note.signatures[0].endsWith('/AnonReviewer.*')
            mailto: Authors: "A reviewer replied a comment from a paper your are a co-author. "
            mailto: Reviewers: "A reviewer replied a comment from a paper your are reviewing. "
            mailto: AreaChairs: "A reviewer replied a comment from a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "A reviewer replied a comment from a paper. "

      if note.signatures[0].endsWith('/Area_Chairs')
            mailto: Authors: "An area chair replied a comment from a paper your are a co-author. "
            mailto: Reviewers: "An area chair replied a comment from a paper your are reviewing. "
            mailto: AreaChairs: "An area chair replied a comment from a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "An area chair replied a comment from a paper. "

      if note.signatures[0].endsWith('/Program_Chairs')
            mailto: Authors: "A program chair replied a comment from a paper your are a co-author. "
            mailto: Reviewers: "A program chair replied a comment from a paper your are reviewing. "
            mailto: AreaChairs: "A program chair replied a comment from a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "A program chair replied a comment from a paper. "

      else 
            mailto: Authors: "An user replied a comment from a paper your are a co-author. "
            mailto: Reviewers: "An user replied a comment from a paper your are reviewing. "
            mailto: AreaChairs: "An user replied a comment from a paper of your are serving as Area Chair."
            mailto: ProgramChairs: "An user replied a comment from a comment to a paper. "

```

Please check the wording @mspector @mohituniyal

Comment (melisabok) [reply venues/ICLR.cc/2019/Conference/process/commentProcess.js]: In the last if I would like to use the ```prettyId(noteReply.invitation)``` to specify if the comment replies to an Official Comment, Public Comment or Official Review.

Comment (melisabok) [reply venues/ICLR.cc/2019/Conference/process/commentProcess.js]: Like: 'An author replied to an Official Review of a paper where your are reviewing'

Comment (mspector) [reply venues/ICLR.cc/2019/Conference/process/commentProcess.js]: oops, sorry I missed this! @melisabok I'm going to edit the text above directly, if that's ok

Comment (mspector) [reply venues/ICLR.cc/2019/Conference/process/commentProcess.js]: ok, I edited the above with what I think is good wording

Comment (melisabok) [reply venues/ICLR.cc/2019/Conference/process/commentProcess.js]: Thanks Michael!

**Comment by zbialecki** [comment]
What's the status of this PR? Do Melisa's suggestions still need to be implemented?

**Comment by mohituniyal** [comment]
@zbialecki yeah, this is still not implemented.

**Comment by melisabok** [comment]
I close this PR and I plan to implement this rule to send notifications in the process function.

**Pull request closed without merging**